plugin/storage/es/factory.go

```diff
@@ -83,6 +83,23 @@ func NewFactory() *Factory {
 	}
 }
 
+func NewFactoryWithConfig(
+	cfg config.Configuration,
+	metricsFactory metrics.Factory,
+	logger *zap.Logger,
+) (*Factory, error) {
+	f := NewFactory()
+	f.InitFromOptions(Options{
+		Primary: namespaceConfig{Configuration: cfg},
+		others:  make(map[string]*namespaceConfig),
```

Comment:
I wonder, if only Primary and not Archive is ever initialized, would it actually provide an archive storage to the UI? Did you try it out?

Reply:
yes, you are right it dose not work with ui. it throw's this error 
this docker container used to test it  
```
docker run --name es01 -p 9200:9200 -e "discovery.type=single-node" -e "xpack.security.enabled=false" elasticsearch:8.12.0
```
![image](https://github.com/jaegertracing/jaeger/assets/122517264/8168b906-e69c-4d22-b439-18be79791e2f)
do you know what causes it?🤔 , i tried fixing it with different configuration of `Options` but couldn't fix.

Reply:
Better to look in the server logs.

Reply:
Can you put together a docker-compose that illustrates this issue?

This "all shards failed" error has always annoyed me, I wonder if there are more details in the error returned from ES that we're not logging that could be more descriptive. Like, it's ridiculous for a database to respond with "your query failed" instead of telling why (the fact that it failed on "all shards" is not an explanation of why).

Reply:
>Can you put together a docker-compose that illustrates this issue?

sure, i'll make one.

Reply:
so when I run v2 with your new config file (and without `-tags ui`), then placeholder UI shows this:
```
JAEGER_CONFIG = {"archiveEnabled":true};
JAEGER_STORAGE_CAPABILITIES = {"archiveStorage":false};
JAEGER_VERSION = {"gitCommit":"","gitVersion":"","buildDate":""};
```
We expect `{"archiveStorage":true}` in order for archiving to work

Reply:
where are we with this one?

Reply:
I didn't check this one previously because there were no errors in the UI. However, I've just reviewed it and found an issue.
```
  function getJaegerStorageCapabilities() {
        const DEFAULT_STORAGE_CAPABILITIES = { "archiveStorage": false };
        const JAEGER_STORAGE_CAPABILITIES = {"archiveStorage":false};
        return JAEGER_STORAGE_CAPABILITIES;
      }
```

Reply:
when i checked with `badger` and `memstore` strangely `{"archiveStorage":false}` in both of them

Reply:
At least for memstore you need to add archive to the config explicitly:
```diff
diff --git a/cmd/jaeger/internal/all-in-one.yaml b/cmd/jaeger/internal/all-in-one.yaml
index d17c52fc..a6189c8e 100644
--- a/cmd/jaeger/internal/all-in-one.yaml
+++ b/cmd/jaeger/internal/all-in-one.yaml
@@ -9,11 +9,14 @@ service:
 extensions:
   jaeger_query:
     trace_storage: memstore
+    trace_storage_archive: memstore2

   jaeger_storage:
     memory:
       memstore:
         max_traces: 100000
+      memstore2:
+        max_traces: 100000
```
But maybe this is ok - the idea in v2 was to decouple primary and archive storage altogether, so that one could use different types of backends for those.

Reply:
So in theory the config you have should've worked, because you did define archive storage, but something isn't working.

Reply:
added a small fix for above problem, which sets `{"archiveEnabled":true}`